_LowLevelCode/CPP/test/cpp_communicator.tests/cpp_communicator_test.cpp

```diff
@@ -0,0 +1,453 @@
+#include "cpp_communicator/MPI_wrapper.h"
+#include "cpp_communicator/input_parser.h"
+#include "cpp_communicator/cpp_communicator.h"
+#include "utility/environment.h"
+
+#include <gtest/gtest.h>
+
+#include <vector>
+
+using namespace Herbert::Utility;
+
+
+TEST(TestCPPCommunicator, send_assynchroneous) {
```

Comment:
Could we try and make these test names more descriptive? The format could be something like `this_does_this_given_this`, it makes it much easier to determine what the test is actually doing.

Reply:
generally agree but its exactly what this test is doing -- testing asynchroneous send given nothing. Is it not clear?
CPPCommunicator_sends_asyncrhoneous_ (no conditions)